### ISSUE TITLE
Reproducing Accuracy on Maze-30x30-hard-1k

### ISSUE THREAD

**Issue by haruki-shirakami** [SamsungSAILMontreal/TinyRecursiveModels#29]
Thanks for the interesting work!

Has anyone been able to reproduce the results on Maze-30x30-hard-1k with the recommended training setup? At global_batch_size=384 (half of recommended) I get pretty large variance validating on the test set during training for 50k epochs (default). 

<img width="5056" height="2656" alt="Image" src="https://github.com/user-attachments/assets/c8aafcc6-ae52-4c46-8d39-4d3547341f99" />

**Comment by gaoxin492** [comment]
@haruki-shirakami yeah, I've reproduced the results: https://github.com/gaoxin492/TinyRecursiveModels

**Comment by olivkoch** [comment]
Yes, making inference on unseen puzzles requires retraining. Fine-tuning TRM is tricky but doable. See https://trelis.substack.com/p/test-time-adaptation-of-tiny-recursive

**Comment by olivkoch** [comment]
I reproduce Maze Hard with this implementation: https://github.com/olivkoch/nano-trm

Just follow the instructions

<img width="454" height="312" alt="Image" src="https://github.com/user-attachments/assets/f253ec55-6f9e-4c80-9575-3866f8b794dc" />
